lib/python/operations.cpp

```diff
@@ -93,6 +94,35 @@ void init_operations(py::module &m) {
   bind_allsorted(m);
   bind_midpoints(m);
 
+  m.def(
+      "label_based_index_to_positional_index",
+      [](const std::vector<std::string> &dims,
+         const std::vector<scipp::index> &shape, const Variable &coord,
+         const Variable &value) {
+        const auto [dim, index] =
+            get_slice_params(make_dims(dims, shape), coord, value);
+        return std::tuple{dim.name(), index};
+      },
+      py::call_guard<py::gil_scoped_release>());
+  m.def("label_based_index_to_positional_index",
+        [](const std::vector<std::string> &dims,
+           const std::vector<scipp::index> &shape, const Variable &coord,
+           const py::slice &py_slice) {
+          auto start = py::getattr(py_slice, "start");
+          auto stop = py::getattr(py_slice, "stop");
+          auto step = py::getattr(py_slice, "step");
+          if (!step.is_none()) {
+            throw std::runtime_error(
+                "Step cannot be specified for value based slicing.");
+          }
+          auto start_var =
+              start.is_none() ? Variable{} : start.cast<Variable>();
+          auto stop_var = stop.is_none() ? Variable{} : stop.cast<Variable>();
+          const auto [dim, start_ind, stop_ind] = get_slice_params(
+              make_dims(dims, shape), coord, start_var, stop_var);
+          return std::tuple{dim.name(), start_ind, stop_ind};
```

Comment:
Didn't this code exist elsewhere?

Reply:
Yes it does. Like the commit message says it is reused. I thought this is what you suggested in the comment above where you wrote.

> Why can't we use that function here (or at least the part inside the try block)

Reply:
"Reuse" means: Reuse the function. If it is a part of a larger function, extract a function. We should never copy code without a good reason (except maybe unit-test setup).